packages/appium/test/driver-e2e-specs.js

```diff
@@ -81,6 +88,53 @@ describe('FakeDriver - via HTTP', function () {
     });
   });
 
+  describe('cli args handling', function () {
+    it('should not recieve user cli args if no cli args passed in', async function () {
+      let driver = await wdio({...wdOpts, capabilities: caps});
+      const {sessionId} = driver;
+      try {
+        const {data} = await axios.get(`${baseUrl}/${sessionId}/fakedriverargs`);
+        data.value.should.eql({});
+      } finally {
+        await driver.deleteSession();
+      }
+    });
+  });
+
+  describe('cli args handling for empty args', function () {
+    it('should recieve user cli args from a driver if arguments were passed in', async function () {
+      await serverClose();
+      let args = {port: TEST_PORT, host: TEST_HOST, appiumHome, driverArgs: FAKE_DRIVER_ARGS};
+      await serverStart(args);
+
+      let driver = await wdio({...wdOpts, capabilities: caps});
+      const {sessionId} = driver;
+      try {
+        const {data} = await axios.get(`${baseUrl}/${sessionId}/fakedriverargs`);
+        data.value.should.eql(FAKE_ARGS);
+      } finally {
+        await driver.deleteSession();
+      }
+    });
+  });
+
+  describe('cli args handling for different driver', function () {
+    it('should recieve empty user cli args from a driver if cli arguments created for a different driver', async function () {
+      await serverClose();
+      let args = {port: TEST_PORT, host: TEST_HOST, appiumHome, driverArgs: `{"xcuitest": ${JSON.stringify(FAKE_ARGS)}}`};
```

Comment:
this would make more sense as:

```js
let args = {..., driverArgs: JSON.stringify({xcuitest: FAKE_ARGS})}
```

Reply:
also, in general i see you using `let` a lot when you're not actually changing the value of the variable later on. our rule is to basically use `const` in every case. if you find you need to change the value of a variable, then you can go back up and change it to `let`.